### PR TITLE
Remaining Voice API Ruby Snippets

### PR DESCRIPTION
**Pull request by benhayehudi**
Added snippets for:

* Track NCCO progress
* Make an outbound call with NCCO

### CODE REVIEW

**Comment by benhayehudi** [comment]
@mheap, Would appreciate any review on this. Thank you :)

File: voice/make-outbound-call-with-ncco.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+client = Nexmo::Client.new(
+  application_id: NEXMO_APPLICATION_ID,
+  private_key: File.read(NEXMO_APPLICATION_PRIVATE_KEY_PATH)
+)
+
+client.calls.create(
+  to: [{
+    type: 'phone',
+    number: RECIPIENT_NUMBER
```
Comment (mheap) [inline voice/make-outbound-call-with-ncco.rb]: I get the following error when trying to run the code:

```
Traceback (most recent call last):
voice/make-outbound-call-with-ncco.rb:20:in `<main>': uninitialized constant RECIPIENT_NUMBER (NameError)
```

Comment (benhayehudi) [reply voice/make-outbound-call-with-ncco.rb]: This has been addressed, thanks for that!

File: voice/track-ncco-progress.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require 'sinatra'
+require 'sinatra/multi_route'
+require 'json'
+
+YOUR_URL = 'put your externally accessible url here'
```
Comment (mheap) [inline voice/track-ncco-progress.rb]: This should be dynamically constructed. See this example from the PHP snippet:

```
            'eventUrl' => [
                $uri->getScheme().'://'.$uri->getHost().':'.$uri->getPort().'/webhooks/notification'
            ]
```

File: voice/track-ncco-progress.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+route :get, :post, '/webhooks/answer' do
+  [
+    [
+      'action' => 'talk',
+      'text' => 'Thanks for calling the notification line'
+    ],
+    [
+      'action' => 'notify',
+      'payload' => ['foo' => 'bar'],
+      'eventUrl' => ["#{YOUR_URL}/webhooks/notifications"]
```
Comment (mheap) [inline voice/track-ncco-progress.rb]: `/notification` not `/notifications`

File: voice/track-ncco-progress.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  puts params
+  [
+    [
+      'action' => 'talk',
+      'text' => 'Your notification has been received, loud and clear'
+    ]
+  ].to_json
+end
+
+route :get, :post, 'webhooks/event' do
```
Comment (mheap) [inline voice/track-ncco-progress.rb]: Missing leading slash?

File: voice/track-ncco-progress.rb (hunk 3 above)
Comment (benhayehudi) [reply voice/track-ncco-progress.rb]: Why? My route on line 29 is defined as `/webhooks/notifications/`

File: voice/track-ncco-progress.rb (hunk 4 above)
Comment (benhayehudi) [reply voice/track-ncco-progress.rb]: 👍

File: voice/track-ncco-progress.rb (hunk 2 above)
Comment (benhayehudi) [reply voice/track-ncco-progress.rb]: 👍

File: voice/track-ncco-progress.rb (hunk 3 above)
Comment (mheap) [reply voice/track-ncco-progress.rb]: For consistency with the existing PHP snippet

Comment (benhayehudi) [reply voice/track-ncco-progress.rb]: Got it. Thanks, it's good to know the reason behind it.

File: voice/track-ncco-progress.rb [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+require 'sinatra'
+require 'sinatra/multi_route'
+require 'json'
+
+before do
+  content_type :json
+end
+
+route :get, :post, '/webhooks/answer' do
```
Comment (mheap) [inline voice/track-ncco-progress.rb]: The NCCO returned here is invalid (note the extra array around each action)

```
curl localhost:3000/webhooks/answer
[[{"action":"talk","text":"Thanks for calling the notification line"}],[{"action":"notify","payload":[{"foo":"bar"}],"eventUrl":["http://localhost:3000/webhooks/notification"]}],[{"action":"talk","text":"You will never hear me as the notification URL will return an NCCO"}]]
```

**Comment by benhayehudi** [comment]
@mheap - can i get another review, please? 🙏

File: voice/track-ncco-progress.rb [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    },
+    {
+      'action' => 'talk',
+      'text' => 'You will never hear me as the notification URL will return an NCCO'
+    }
+  ].to_json
+end
+
+route :get, :post, '/webhooks/notification' do
+  puts params
```
Comment (mheap) [inline voice/track-ncco-progress.rb]: This is empty in the server log

```
169.63.86.171 - - [17/Jul/2019:17:41:53 +0100] "POST /webhooks/notification HTTP/1.1" 200 80 0.0015
{}
```

File: voice/track-ncco-progress.rb [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  [
+    {
+      'action' => 'talk',
+      'text' => 'Your notification has been received, loud and clear'
+    }
+  ].to_json
+end
+
+route :get, :post, '/webhooks/event' do
+  puts params
```
Comment (mheap) [inline voice/track-ncco-progress.rb]: This is also empty

```
169.63.86.171 - - [17/Jul/2019:17:41:38 +0100] "POST /webhooks/event HTTP/1.1" 204 - 0.0013
{}
```

File: voice/track-ncco-progress.rb [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+route :get, :post, '/webhooks/answer' do
+  [
+    {
+      'action' => 'talk',
+      'text' => 'Thanks for calling the notification line'
+    },
+    {
+      'action' => 'notify',
+      'payload' => ['foo' => 'bar'],
```
Comment (mheap) [inline voice/track-ncco-progress.rb]: This should be `{'foo' => 'bar'}` I believe

File: voice/make-outbound-call-with-ncco.rb [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+  }],
+  from: {
+    type: 'phone',
+    number: NEXMO_NUMBER
+  },
+  ncco: [
+    'action' => 'talk',
```
Comment (mheap) [inline voice/make-outbound-call-with-ncco.rb]: Could we add braces around the action to make it explicit that it's a list of objects?

File: voice/track-ncco-progress.rb [hunk 10]
```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require 'sinatra'
+require 'sinatra/multi_route'
+require 'rack/contrib'
```
Comment (fabianrbz) [inline voice/track-ncco-progress.rb]: should these dependencies be listed in the `Gemfile`?

**Pull request merged**